### ISSUE TITLE
Inconsistency saving after 'AddRegion' and 'SubstractRegion'

### ISSUE THREAD

**Issue by celinede** [r03ert0/microdraw#193]
Hi,

I have noticed another inconsistency when saving the data.

After using the 'AddRegion' tool the new region is saved with a different format, using the key word 'CompoundPath' and 'children' in the json. Microdraw is afterwards not able to draw the region anymore, while the region is still saved in the database and shown in the list of Regions.
One way that I have found to correct this problem of formatting is to edit anything on the section and save which will update the formatting the problematic region. After refreshing the page the region reappears.
![Screenshot from 2019-05-07 14-47-00](https://user-images.githubusercontent.com/18595712/57300407-0b37da00-70d7-11e9-8b5e-53bc485afb33.png)

Exactly the same issue happens after using the 'SubstractRegion' tool.
![Screenshot from 2019-05-07 14-40-40](https://user-images.githubusercontent.com/18595712/57300113-687f5b80-70d6-11e9-9e9a-36be5d1d53bc.png)

Thanks for your help !

**Comment by xgui3783** [comment]
ok, the weirdest thing happened. I could reproduce the first issue (compound path) at first. But every subsequent attempt I failed at reproducing the issue. 

Here is how I tried to reproduce the issue:

1/ use draw region tool (draw poly worked similarly) to draw two regions
2/ I select one of the regions, select add region tool, select the other region (at this point, the regions should union, and there should only be one reason visible)
3/ click save, and inspect the network payload. 

Which tools are you using? maybe that can help us narrow down the issue.

**Comment by celinede** [comment]
That's exactly what I do.
I use only polygons at the moment.
After saving, the problem occurs after refreshing the page.

I think I've spotted what it doesnt like : if when adding the second region
it leaves a hole in the contour. Or if there is a loop in one of the
contours. Which gives in the end some kind of torus shape.
That leads to the compound path format which can't be displayed.
Then if you edit and save something on the section and refresh again, the
format will be corrected and the shape will be edited in order to have a
plain shape (removes any kind of hole or loop).

This has actually been a big problem that I have had lately. I had worked
on delineating the sulci of my ferret brains (which leads to overlapping
lines because the sulci are very tight) and suddenly all my corrections in
the sulci were lost. I guess that comes from the same issue.

In my case it would be good if overlapping lines/loops could be allowed or
controlled so that I don't expectedly loose some work. My problem is that
in sulci, I need to go down the sulci and back up more or less on the same
line depending on how tight it is and it's not easy to see if there is an
overlap...

Thanks so much for your help!
